### DISCUSSION TITLE
Where does LazyVim set its runtime paths?

### DISCUSSION THREAD

**Discussion by nianiam** [LazyVim/LazyVim#1464] [General]
First of all, thank you for all your hard work making these tools, I learnt a lot from them.

I was using LazyVim until recently, but found that I was disabling/overriding many of the plugins to the the point that it made more sense to just build my own. I am still using `lazy.nvim` but found that the `ftplugin` and `after/*` folders would not work by default. After some investigation I found that I needed to set them explicitly in lazy's config for them to be sourced:

```lua
{
  performance = {
    ...
    rtp = {
      ...
      paths = {
        "~/<User>/.config/nvim/lua/after/",
        "~/<User>/.config/nvim/lua/ftplugin",
      },
    },
  },
}
```

However, when running LazyVim these would be sourced automatically. My question is where does LazyVim set these values? I would like to be able to avoid using explicit paths if I can dynamically define them during loading.

**Comment by abeldekat** [comment]
Hi @nianiam,

I would expect the `after` and `ftplugin` folders to be located in the top level of your `neovim` config, not in the `lua` folder.

You don't need to explicitly add them to the rtp in `lazy.nvim`'s `setup`.

Best regards!

**Reply by dpetka2001** [reply]
Like @abeldekat mentioned, they should be in the top level of your neovim config, as every directory under `/lua/` subdirectory is considered a Lua module which should be `require`d.

**Reply by abeldekat** [reply]
By the way, out of curiosity, how many plugins did you disable? I currently disabled 9 plugins. I consider `LazyVim` a part of my own config though, because I understand the code as I would understand my own.

**Reply by nianiam** [reply]
You guys are right, moving them out and lazy sourced them properly... I guess I need to learn a bit more! Thanks.

@abeldekat I disabled 4-5, and replaced them with other similar plugins. However, I re-configured well over 15 plugins of the included plugins so it was getting to the point where I may as well just do it myself. Additionally, I didn't like being (albeit in good hands) beholden to a package with so many dependencies. I also found LazyVim to be a little bit laggy as compared to a lighter setup, although this might be workflow-specific/subjective.

With all that being said, I am incredibly grateful LazyVim exists and would still recommend it to people starting out with neovim.
